src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt

```diff
@@ -15,14 +16,26 @@ class DetailService(private val sqlRepository: SqlRepository, val authentication
     fun getStaffDetails(from: LocalDate, to: LocalDate, quantumId: String = authenticationFacade.currentUsername): Collection<DetailDto> {
         log.debug("Fetching shift details for $quantumId")
         val details = sqlRepository.getDetails(from, to, quantumId)
-        log.info("Found ${details.size} shift details for $quantumId")
+        val templateNames = getTemplatesSet(details).toList()
+        val templates = if(templateNames.any()) {
+             sqlRepository.getDetailTemplates(templateNames)
+        } else {
+            listOf()
+        }
+
+        val mergedDetails = mergeTemplatesIntoDetails(details, templates)
+
 
-        return mapToDetailsDto(details)
+        log.info("Found ${mergedDetails.size} shift details for $quantumId")
+
+        return mapToDetailsDto(mergedDetails)
     }
 
     fun getModifiedDetailsByPlanUnit(planUnit: String): Collection<DetailDto> {
         log.debug("Fetching modified shifts for $planUnit")
         val modifiedShifts = sqlRepository.getModifiedShifts(planUnit)
+
+        //TODO: add in template data
```

Comment:
@nickmcmahon01 do we wanna do this? I think we'd need it for notifications, but you've probably got a better idea of how that side fits together

Reply:
I'm not sure, and to be blunt I'm not sure the team would know either. lets ignore it for now as its missing now in production (any currently doesn't use the plugin which this work replaces). I'll look into it.